File: Data Structures & Algorithms/islands-and-treasure/submission-0.cpp
class Solution {
public:
    void islandsAndTreasure(vector<vector<int>>& grid) {
        int cols = grid[0].size();
        int rows = grid.size();
        vector<vector<int>> visited(rows,vector<int>(cols,0));
        queue<pair<int,int>> q;
        for(int r = 0;r< rows;r++){
            for(int c =0; c< cols;c++){
                if(grid[r][c] ==0){
                    q.push({r,c});
                }
            }
        }
        int dist = 0;
        while(!q.empty()){
            int n = q.size();
             dist+=1;
            for(int i=0;i< n;i++){
                    auto x = q.front();
                    q.pop();
                    int  r = x.first;
                    int c = x.second;
                    visited[r][c] = 1;
                    vector<pair<int,int>> directions = {{0,1},{0,-1},{1,0},{-1,0}};
                    for(auto dir : directions){
                        int rr = dir.first+ r;
                        int cc = dir.second+ c;
                        if(rr < rows && cc < cols && rr >=0 && cc >=0 
                        && visited[rr][cc]==0 && grid[rr][cc]!=-1){
                            visited[rr][cc]=1;
                            grid[rr][cc]= dist;
                            q.push({rr,cc});
                        }
                    }
            }
           
        }
    }
    
};

File: Data Structures & Algorithms/set-zeroes-in-matrix/submission-0.py
class Solution:
    def setZeroes(self, matrix: List[List[int]]) -> None:
        M = len(matrix)
        N = len(matrix[0])

        cols = [0]* N
        rows = [0]*M
        for i in range(M):
            for j in range(N):
                if matrix[i][j] == 0:
                    cols[j] =1 
                    rows[i] = 1
        
        for i in range(M):
            for j in range(N):
                if cols[j] ==1 or rows[i] == 1:
                    matrix[i][j] = 0
        